docs/deployment/uwsgi_cookiecutter_nginx.rst

```diff
@@ -0,0 +1,233 @@
+to which many Python web frameworks conform).
+
+This guide will outline broad steps that can be used to get a Cookiecutter
+:app:`Pyramid` application running under Nginx via ``uWSGI``.  This particular
+tutorial was developed under Ubuntu 18.04, but the instructions should be
+largely the same for all systems, delta specific path information for commands
+and files.
+
+
+#.  Install prerequisites
+
+    .. code-block:: bash
+
+       $ sudo apt install -y uwsgi-core uwsgi-plugin-python3 python3-cookiecutter \
+                             python3-pip python3-venv nginx
+       $ python3 -m cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+
+#.  Create a :app:`Pyramid` application. For this tutorial we'll use the
+    ``starter`` :term:`cookiecutter`. See :ref:`project_narr` for more
+    in-depth information about creating a new project.
+
+    .. code-block:: bash
+
+       $ cd
+       $ cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout master
+
+    If prompted for the first item, accept the default ``yes`` by hitting return.
+
+    .. code-block:: text
+
+        You've cloned ~/.cookiecutters/pyramid-cookiecutter-starter before.
+        Is it okay to delete and re-clone it? [yes]: yes
+        project_name [Pyramid Scaffold]: myproject
+        repo_name [myproject]: myproject
+        Select template_language:
+        1 - jinja2
+        2 - chameleon
+        3 - mako
+        Choose from 1, 2, 3 [1]: 1
+
+#.  Create a :term:`virtual environment` which we'll use to install our
+    application.
+
+    .. code-block:: bash
+
+       $ cd myproject
+       $ python3 -m venv env
+
+#.  Install your :app:`Pyramid` application and its dependencies.
+
+    .. code-block:: bash
+
+       $ env/bin/pip install -e ".[testing]"
+
+#.  Within the project directory (``~/myproject``), create a script
+    named ``wsgi.py``.  Give it these contents:
+
+    .. code-block:: python
+
+        # Adapted from PServeCommand.run in site-packages/pyramid/scripts/pserve.py
+        from pyramid.scripts.common import get_config_loader
+        app_name    = 'main'
+        config_vars = {}
+        config_uri  = 'production.ini'
+
+        loader = get_config_loader(config_uri)
+        loader.setup_logging(config_vars)
+        app = loader.get_wsgi_app(app_name, config_vars)
+
+    `config_uri` is the project configuration file name.  It's best to use
+    the ``production.ini`` file provided by your cookiecutter, as it contains
+    settings appropriate for production.  `app_name` is the name of the section
+    within the ``.ini`` file that should be loaded by ``uWSGI``.  The
+    assignment to the name ``app`` is important: we will reference ``app`` and
+    the name of the file, ``wsgi`` when we invoke uWSGI.
+
+    The call to :func:`loader.setup_logging` initializes the standard
+    library's `logging` module to allow logging within your application.
+    See :ref:`logging_config`.
```

Comment:
Although linking via intersphinx will work, I suggest to use explicit prefixes with the following modifications:

```
The call to ``loader.setup_logging`` initializes the standard
library's :mod:`python3:logging` module through :func:`pyramid.paster.setup_logging` to  to allow logging within your application. See :ref:`pyramid:logging_config`.
```

Reply:
That's much more detailed.